### PR TITLE
Navigator: Add optional navigationBarPosition prop

### PR DESCRIPTION
**Pull request by arvindch**
When passing a custom `navigationBar`, like `ToolbarAndroid`, we need a way to ensure the toolbar is rendered at the top of the page. This makes sure we don't need any CSS tricks or hacks to render it.

**Test plan (required)**

Default is still `bottom`, which means no tests should be affected.

It's a simple change, `navigationBar` is rendered before the `View` in `Navigator`, to ensure it shows up at the `top`.

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
@arvindch updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9839/files/33a0b6ff2ad43d6dea6a8a993c9b2873f0750f4d..8156feb72a644af989e1cc2d1577763026315346)

**Comment by facebook-github-bot-3** [comment]
@arvindch updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9839/files/8156feb72a644af989e1cc2d1577763026315346..3b5e5ac3806c2696e69610f799b28aaffa845646)

File: Libraries/CustomComponents/Navigator/Navigator.js [hunk 1]
```diff
@@ -366,6 +366,12 @@ var Navigator = React.createClass({
     navigationBar: PropTypes.node,
 
     /**
+     * Optionally choose the position to render navigation bar at.
+     * It defaults to 'bottom'
+     */
+    navigationBarPosition: PropTypes.oneOf(['bottom' ,'top']),
```
Comment (hramos) [inline Libraries/CustomComponents/Navigator/Navigator.js]: Why is the default behavior 'bottom'?

**Comment by ericvicenti** [comment]
Instead of doing this, you can use absolute positioning to make sure the header is displayed above the navigator. Set the height, and, `{position: 'absolute', top: 0, left: 0, right: 0}`

File: Libraries/CustomComponents/Navigator/Navigator.js (hunk 1 above)
Comment (arvindch) [inline Libraries/CustomComponents/Navigator/Navigator.js]: Because right now it always renders to bottom. It's already the default.

**Comment by arvindch** [comment]
@ericvicenti - Yes, I've tried that. It results in the navbar overlapping content at the top.

Seeing as how react-native is supposed to be cross-platform, and Android uses navbar at the top, shouldn't this be an obvious feature? I don't the the code changes I proposed are intrusive or major in any way, either.

**Comment by NomadGraphix** [comment]
Please reopen this. It seems un-intuitive to abstract away the physical positioning of react components. I don't necessarily like the 'binary' idea of providing either 'top' or 'bottom' to fix this conundrum, so I've placed my suggestion below.

My suggestion is to instead create an alternate 'higher-level' render prop to the <Navigator /> component. 

For example: 

```javascript

<Navigator
   renderScene={(route, navigator) => <Text>Test</Text>}
   navigationBar={<MyCustomNavbar />}
   renderNavigatorScene={(Navbar, Scene) => {
       return (
           <View>
              {/* Customize component order and insert your own additional components at this level. */}
              <Navbar /> {/* Component with injected navigator, navState props */}
              <Scene /> {/* Component with injected route, navigator props */}
           </View>   
       )
   }}
/>
```

**Pull request closed without merging**